### PR TITLE
[IP2P] Make text encoder truly optional in InstructPi2Pix

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

Having this optional could really help use cases like [MGIE](https://github.com/apple/ml-mgie) to save further memory.

### CODE REVIEW

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_instruct_pix2pix.py [hunk 1]
```diff
@@ -553,13 +553,17 @@ def _encode_prompt(
...
+            prompt_embeds = self.text_encoder(text_input_ids.to(device), attention_mask=attention_mask)
             prompt_embeds = prompt_embeds[0]
 
-        prompt_embeds = prompt_embeds.to(dtype=self.text_encoder.dtype, device=device)
+        if self.text_encoder is not None:
+            prompt_embeds_dtype = self.text_encoder.dtype
+        elif self.unet is not None:
+            prompt_embeds_dtype = self.unet.dtype
+        else:
+            prompt_embeds_dtype = prompt_embeds.dtype
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_instruct_pix2pix.py:564]: it is not possible for unet to be `None`, no? 
```suggestion
        else:
            prompt_embeds_dtype = self.unet.dtype
```

**Review by yiyixuxu — APPROVED** [review]
Great!
I imagine this benefits all other pipelines accepts `prompt_embeds`, no?

**Comment by sayakpaul** [comment]
> I imagine this benefits all other pipelines accepts prompt_embeds, no?

Yes, how about I tackle a few of the major ones and then open it to the community? ("Tracker" issue)

**Comment by yiyixuxu** [comment]
@sayakpaul sounds good!

**Pull request merged**